### ISSUE TITLE
[GH Issue Summarization] HP Controller should launch TFJobs directly

### ISSUE THREAD

**Issue by jlewi** [kubeflow/examples#162]
#161 Moved over the HP controller program github-issue-summarize-demo.go from Katib to the examples repository.

The jobs launched currently fail because we aren't specifying a GCP credential which is needed to download the data. 

One solution would be to modify the controller program so that we set the relevant mounts in WorkerConfig to mount the controller.

I'm wondering though if a better solution would be for the controller to directly launch the jobs by creating  TFJobs rather than calling katib's manager.RunTrial function?

As far as I can tell [manager.RunTrial](https://github.com/kubeflow/katib/blob/master/cmd/manager/main.go#L107) doesn't do much besides issuing the Create Job request to K8s master. Getting the suggested parameters and reporting metrics are separate calls in the controller program.

I think the advantage of creating the K8s resources directly is that the controller can determine how to create the K8s resource which gives us a lot more flexibility and control.

In particular, it would be nice to reuse the existing ksonnet templates e.g [tf-job-v1alpha2](https://github.com/kubeflow/examples/blob/master/github_issue_summarization/ks-kubeflow/components/tfjob-v1alpha2.jsonnet) and ksonnet for injecting the hyperparameters into the template. 

So the controller could basically do the equivalent of 

  * ks param set learning_rate ....
  * ks apply

to create the resources.

**Comment by gaocegege** [comment]
manager.RunTrial allows us to use different worker interface, including tfjob or k8s directly. I think manager is designed to do what you want to do:

> the advantage of creating the K8s resources directly is that the controller can determine how to create the K8s resource which gives us a lot more flexibility and control.

/cc @YujiOshima
